frontend/www/js/omegaup/graderv2/CaseSelectorComponent.vue

```diff
@@ -0,0 +1,306 @@
+        <button
+          v-if="!groups"
+          class="list-group-item list-group-item-action disabled"
+          type="button"
+        >
+          <em>Empty</em>
+        </button>
+        <template v-for="group in groups" v-else :title="name">
+          <div
+            v-if="group.explicit"
+            :key="group.name"
+            class="list-group-item list-group-item-secondary"
+          >
+            <div>
+              <span
+                class="verdict"
+                :class="verdictClass(groupResult(group.name))"
+                :title="verdictTooltip(groupResult(group.name))"
+                >{{ verdictLabel(groupResult(group.name))
+                }}<span class="score">{{
+                  score(groupResult(group.name))
+                }}</span></span
+              >
+              <span :title="group.name">{{ group.name }}</span>
+            </div>
+          </div>
+          <button
+            v-for="item in group.cases"
+            :key="item.name"
+            class="list-group-item list-group-item-action d-flex justify-content-between align-items-center"
+            type="button"
+            :class="{
+              'in-group': group.explicit,
+              active: currentCase == item.name,
+            }"
+            @click="selectCase(item.name)"
+          >
+            <div class="case-item">
+              <span
+                class="verdict"
+                :class="verdictClass(item.name)"
+                :title="verdictTooltip(caseResult(item.name))"
+                >{{ verdictLabel(caseResult(item.name))
+                }}<span class="score">{{
+                  score(caseResult(item.name))
+                }}</span></span
+              >
+              <span :title="item.name">{{ item.name }}</span>
+            </div>
+            <button
+              v-if="groups.length > 1"
+              aria-label="Close"
+              class="close"
+              type="button"
+              @click.prevent.stop="removeCase(item.name)"
+            >
+              <span aria-hidden="true">×</span>
+            </button>
+          </button>
+        </template>
+      </div>
+    </div>
+    <form @submit.prevent="createCase()">
+      <div class="input-group">
+        <input
+          v-model="newCaseWeight"
+          class="form-control case-weight"
+          type="text"
+        />
+        <input v-model="newCaseName" class="form-control" type="text" />
+        <div class="input-group-append">
+          <button
+            class="btn btn-secondary"
+            type="submit"
+            :disabled="newCaseName.length == 1"
+            @click="createCase()"
+          >
+            +
+          </button>
+        </div>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import * as Util from '../grader/util';
+
+@Component
+export default class MyComponent extends Vue {
```

Comment:
nit:
```suggestion
export default class GraderCaseSelector extends Vue {
```

Additionally, all the `.vue` files are components. Is it possible to remove the `Component` suffix from the filenames?